Lib/base64.py

```diff
@@ -601,7 +601,17 @@ def main():
         with open(args[0], 'rb') as f:
             func(f, sys.stdout.buffer)
     else:
-        func(sys.stdin.buffer, sys.stdout.buffer)
+        if sys.stdin.isatty():
+            # gh-gh-138775: read input data at once when reading from stdin
+            # This allows proper handling of EOF (Ctrl+D)
+            input_data = sys.stdin.buffer.read()
+            if input_data:
+                import io
+                input_buffer = io.BytesIO(input_data)
+                func(input_buffer, sys.stdout.buffer)
```

Comment:
```suggestion
            import io
            data = sys.stdin.buffer.read()
            func(io.BytesIO(data), sys.stdout.buffer)
```

We should still return b'' for the empty string.

Reply:
got it